### ISSUE TITLE
Failed to negotiate cipher with server on v.0.7.19

### ISSUE THREAD

**Issue by omid-1985** [schwabe/ics-openvpn#1232]
# General information
1. Android Version 10
2. SamsungOne UI 2.1
3. Galaxy S10+
4. Version of the app 0.7.19
 
# Description of the issue
Latest version breaks VPN connectivity to my open VPN server. if I roll back to v.0.7.16 the connection starting to work again:

![93691122-6cbde480-fb24-11ea-838b-9f250bb8c105](https://user-images.githubusercontent.com/19699810/97655596-9a9a2f80-1ab9-11eb-96d3-cc1b0d62d846.jpg)

![2](https://user-images.githubusercontent.com/19699810/97655696-cae1ce00-1ab9-11eb-9caa-32ac8dc51ff9.jpeg)

**Comment by schwabe** [comment]
Has the config been import in 0.7.19 or was it an upgrade? And you could paste the config (without private keys of coruse)

**Comment by omid-1985** [comment]
The configuration is the same as what I had in the previous version, I did not re-import or change anything, just app upgrade.

**Comment by heyzec** [comment]
Same issue here. Issue appears after latest update. Did not change configurations.

**Comment by schwabe** [comment]
@omid-1985, @heyzec I still need to understand why it is failing for you. And for that I need log and configuration profiles. OpenVPN 2.5 and the upcoming 2.6 made changes to cipher negotiation and to debug where the problem I need to understand it. So it would be good in your own interest to send me the files.

**Comment by w6gw8h** [comment]
I had the same problem after updating from 0.7.17a to 0.7.18 (F-Droid).

The client log on Android told me that the cipher was expected to be AES-256-GCM, but the server only offered AES-256-CBC. (I admit that I have to update my server...)

The following worked for me:

* Start your Android OpenVPN client.
* Select your profile (the pencil).
* Select tab "extended" ("ERWEITERT" in German version).
* Activate "own configuration", unless already done.
* Touch "own options" ("Eigene Optionen" in German version).
* Enter `data-ciphers AES-256-CBC`
* Enter `data-ciphers-fallback AES-256-CBC`
* Touch "OK".
* Try reconnection.

The commit entry `Ignore --cipher for cipher negotiation in server client mode` was suspicious for me. ;-)

**Comment by schwabe** [comment]
There might be still a bug. The update is supposed to automatically add your cipher to data-ciphers to avoid these kind of situation. Future OpenVPN version will behave like my app but I think it is a bit too early for that.

**Comment by omid-1985** [comment]
**I have fixed the issue by unticking "Enable Custom Options":**

![1](https://user-images.githubusercontent.com/19699810/97655819-0e3c3c80-1aba-11eb-89bc-457cf16ce39e.jpg)

On "Custom Options" I had: ncp-ciphers AES-128-GCM but my actual encryption method is AES-256-GCM.
I don't remember ever ticking "Enable Custom Options" or writing any custom value on it. 

![2](https://user-images.githubusercontent.com/19699810/97655838-13998700-1aba-11eb-9fe3-c0b6b75444d9.jpg)

**Comment by omid-1985** [comment]
> @omid-1985, @heyzec I still need to understand why it is failing for you. And for that I need log and configuration profiles. OpenVPN 2.5 and the upcoming 2.6 made changes to cipher negotiation and to debug where the problem I need to understand it. So it would be good in your own interest to send me the files.


@schwabe Log was provided on the first post. Here is the first and last part of the configuration profile, the middle section was just the Certificate and Key values:

![20200920_015255](https://user-images.githubusercontent.com/19699810/93671207-6c9df480-fae4-11ea-927e-893b9bae5dab.jpg)
![20200920_015200](https://user-images.githubusercontent.com/19699810/93671209-6dcf2180-fae4-11ea-84b2-2960cba7199c.jpg)

**Comment by w6gw8h** [comment]
> There might be still a bug.

When I opened "own options", there was a comment line (`# The following...` - I don't remember exactly), followed by some inactive options. Maybe, the auto-adaption from `ciper` to `data-ciphers` is not done, if the comment character is present.

It's just an idea. I didn't check the source code.

**Comment by schwabe** [comment]
All options from original config that are not mapped to a setting end up in custom options and ncp-ciphers is not a commonly used option. Where is the original configuration coming from? A specific VPN provider? Some OpenVPN tutorial?

**Comment by omid-1985** [comment]
> All options from original config that are not mapped to a setting end up in custom options and ncp-ciphers is not a commonly used option. Where is the original configuration coming from? A specific VPN provider? Some OpenVPN tutorial?

@schwabe 

Configuration generated from pfSense OpenVPN Server. Found ncp value on it:

![Screenshot 2020-09-20 103124](https://user-images.githubusercontent.com/19699810/93691786-9e877900-fb2d-11ea-960b-1200bafe4852.png)

What I don't understand is why this wasn't an issue on the previous version?

**Comment by schwabe** [comment]
It looks like the authors of pfSense did not really understand what the option is/whas supposed to be used. NCP is not only for old peers for compatibility but actually takes preference over --cipher. It always did. OpenVPN has also added --cipher by default into the list of ncp-cipher in past. But with OpenVPN 2.6 OpenVPN will completely drop --cipher and only evaluate '--data-ciphers' (new name of --ncp-ciphers). 

I am doing some conversion of now adding --cipher to --data-ciphers but some corner cases of old config ncp-ciphers keep the ncp-ciphers intact. This should normally not be a problem because most OpenVPN 2.4 configs that use ncp-ciphers use an identical setting on server and client. 

From the error message from your log ncp is actually disabled on the server, so the client is forced to switch to the cipher of the server but that is not included in the ncp-ciphers setting. 

In summary you are running in the obscure corner case of *explicity* setting ncp-ciphers on client (which the import still honours and does not mangle) and then also using ncp-disable on the server.

**Comment by schwabe** [comment]
I created an issue on the pfSense bugtracker to avoid creating problem like this in the future: https://redmine.pfsense.org/issues/10919

**Comment by wazabees** [comment]
I have the same use case. pfSense OpenVPN server with configuration files generated by [		openvpn-client-export](https://docs.netgate.com/pfsense/en/latest/vpn/openvpn/using-the-openvpn-client-export-package.html), and I'm seeing the same errors. Is there a different way we should be [configuring our OpenVPN servers](https://docs.netgate.com/pfsense/en/latest/book/openvpn/openvpn-configuration-options.html) to be able to use the latest ics-openvpn?

**Comment by schwabe** [comment]
@wazabees I am not even sure how you end up with a mismatched server/client config. Even looking at the provided screenshot I feel that you should end up with the same ncp-disable/ncp-ciphers settings in server and client config. If pfSense exports client and server configs with different ncp-disable/ncp-ciphers settings than you are in trouble. I still not sure what exactly happens since I haven't seen the actual generated client/server configs from pfSense.

**Comment by wazabees** [comment]
> I still not sure what exactly happens since I haven't seen the actual generated client/server configs from pfSense.

@schwabe I can send you a pfSense generated client .ovpn file if you're interested

**Comment by lazmo88** [comment]
Similar issues here with OVPN profile generated by Asus router.

Error is "unsupported cipher in --data-ciphers: none"

Options error: NCP cipher list contains unsupported ciphers or is too long.

Apparently "none" as cipher isn't supported anymore, it was working fine in the previous versions.

Version 0.7.16 works fine.

Also Android TV UI is completely unusable. You can't select between profiles as you could before.

**Comment by DD8372812** [comment]
Same here with OpenVPN Server on IPFire. But still not clear to me what to change.

**Comment by schwabe** [comment]
No encryption is just not supported in the current version, you will have to wait for the next version.

**Comment by lazmo88** [comment]
@schwabe was there a specific reason why support for no encryption was removed?

**Comment by schwabe** [comment]
@lazmo88 no encryption is a feature that normally shouldn't be used at all. And it gets little testing. And therefore it broke. AES-128-GCM is almost as fast on most platform, so there is little reason to use no encryption.

**Comment by lazmo88** [comment]
@schwabe in our use case even the AES-128-GCM created unwanted hit to our throughput. We only use VPN for non-sensitive TV streaming (selected apps, without authentication). So, we are aware of risks involved with no encryption, yet, it is critical feature for us and we do hope it will be brought back.

**Comment by schwabe** [comment]
@lazmo88 as I said, you will have to wait for the next version.

**Comment by oceanBT** [comment]
Hi, 
i have the same error as @omid-1985 . I use the app since 2 years, and have not changed anything in config. On server (ubuntu 16.04) i use openvpn 2.3.10.
When i add `data-ciphers AES-256-CBC ` to custom config, i can connect as usual. 

Maybe this helps finding the bug:

config:


```

# Config for OpenVPN 2.x
# Enables connection to GUI
management /data/user/0/de.blinkt.openvpn/cache/mgmtsocket unix
management-client
management-query-passwords
management-hold

setenv IV_GUI_VER "de.blinkt.openvpn 0.7.18" 
setenv IV_SSO openurl,crtext
setenv IV_PLAT_VER "29 10 arm64-v8a Xiaomi msm8998 Mi MIX 2"
machine-readable-output
allow-recursive-routing
ifconfig-nowarn
client
verb 4
connect-retry 2 300
resolv-retry 60
dev tun
remote XX 1194 udp
<ca>
-----BEGIN CERTIFICATE-----
-----END CERTIFICATE-----
</ca>
<key>
-----BEGIN PRIVATE KEY-----
-----END PRIVATE KEY-----
</key>
<cert>
</cert>
comp-lzo
<tls-auth>
#
# 2048 bit OpenVPN static key
#
-----BEGIN OpenVPN Static key V1-----
-----END OpenVPN Static key V1-----
</tls-auth>
key-direction 1
route XX  255.255.255.0 vpn_gateway
nobind
verify-x509-name ""
remote-cert-tls server
cipher AES-256-CBC
auth SHA512
persist-tun
# persist-tun also enables pre resolving to avoid DNS resolve problem
preresolve
# Use system proxy setting
management-query-proxy
# Custom configuration options
# You are on your on own here :)
# These options found in the config file do not map to config settings:
resolv-retry infinite 
tls-cipher TLS-DHE-RSA-WITH-AES-256-CBC-SHA256:TLS-DHE-RSA-WITH-AES-128-CBC-SHA256:TLS-DHE-DSS-WITH-AES-128-CBC-SHA256:TLS-DHE-RSA-WITH-AES-128-CBC-SHA 


```
my logfile:
```
2020-10-11 14:21:31 F-Droid built and signed version 0.7.18 läuft auf Xiaomi Mi MIX 2 (msm8998), Android 10 (QQ3A.200805.001) API 29, ABI arm64-v8a, (Xiaomi/chiron/chiron:8.0.0/OPR1.170623.027/V9.5.4.0.ODEMIFA:user/release-keys)
2020-10-11 14:21:31 Generiere OpenVPN-Konfiguration…
2020-10-11 14:21:31 started Socket Thread
2020-10-11 14:21:31 P:WARNING: linker: Warning: "/data/app/de.blinkt.openvpn-ZeExaDhF-roAtxJ2zEP-xg==/lib/arm64/libovpnexec.so" is not a directory (ignoring)
2020-10-11 14:21:31 Netzwerkstatus: CONNECTED LTE to MOBILE internet.telekom
2020-10-11 14:21:31 Debug state info: CONNECTED LTE to MOBILE internet.telekom, pause: userPause, shouldbeconnected: true, network: SHOULDBECONNECTED 
2020-10-11 14:21:31 Debug state info: CONNECTED LTE to MOBILE internet.telekom, pause: userPause, shouldbeconnected: true, network: SHOULDBECONNECTED 
2020-10-11 14:21:31 WARNING: Compression for receiving enabled. Compression has been used in the past to break encryption. Sent packets are not compressed unless "allow-compression yes" is also set.
2020-10-11 14:21:31 Note: --cipher set to 'AES-256-CBC' but missing in --data-ciphers (AES-256-GCM:AES-128-GCM). OpenVPN 2.6+ ignores --cipher for cipher negiotiation.
2020-10-11 14:21:31 Current Parameter Settings:
2020-10-11 14:21:31   config = '/data/user/0/de.blinkt.openvpn/cache/android.conf'
2020-10-11 14:21:31   mode = 0
2020-10-11 14:21:31   show_ciphers = DISABLED
2020-10-11 14:21:31   show_digests = DISABLED
2020-10-11 14:21:31   show_engines = DISABLED
2020-10-11 14:21:31   genkey = DISABLED
2020-10-11 14:21:31   genkey_filename = '[UNDEF]'
2020-10-11 14:21:31   key_pass_file = '[UNDEF]'
2020-10-11 14:21:31   show_tls_ciphers = DISABLED
2020-10-11 14:21:31   connect_retry_max = 0
2020-10-11 14:21:31 Connection profiles [0]:
2020-10-11 14:21:31   proto = udp
2020-10-11 14:21:31   local = '[UNDEF]'
2020-10-11 14:21:31   local_port = '[UNDEF]'
2020-10-11 14:21:31   remote = 'XX'
2020-10-11 14:21:31   remote_port = '1194'
2020-10-11 14:21:31   remote_float = DISABLED
2020-10-11 14:21:31   bind_defined = DISABLED
2020-10-11 14:21:31   bind_local = DISABLED
2020-10-11 14:21:31   bind_ipv6_only = DISABLED
2020-10-11 14:21:31   connect_retry_seconds = 2
2020-10-11 14:21:31   connect_timeout = 120
2020-10-11 14:21:31   socks_proxy_server = '[UNDEF]'
2020-10-11 14:21:31   socks_proxy_port = '[UNDEF]'
2020-10-11 14:21:31   tun_mtu = 1500
2020-10-11 14:21:31   tun_mtu_defined = ENABLED
2020-10-11 14:21:31   link_mtu = 1500
2020-10-11 14:21:31   link_mtu_defined = DISABLED
2020-10-11 14:21:31   tun_mtu_extra = 0
2020-10-11 14:21:31   tun_mtu_extra_defined = DISABLED
2020-10-11 14:21:31   mtu_discover_type = -1
2020-10-11 14:21:31   fragment = 0
2020-10-11 14:21:31   mssfix = 1450
2020-10-11 14:21:31   explicit_exit_notification = 0
2020-10-11 14:21:31   tls_auth_file = '[INLINE]'
2020-10-11 14:21:31   key_direction = 1
2020-10-11 14:21:31   tls_crypt_file = '[UNDEF]'
2020-10-11 14:21:31   tls_crypt_v2_file = '[UNDEF]'
2020-10-11 14:21:31 Connection profiles END
2020-10-11 14:21:31   remote_random = DISABLED
2020-10-11 14:21:31   ipchange = '[UNDEF]'
2020-10-11 14:21:31   dev = 'tun'
2020-10-11 14:21:31   dev_type = '[UNDEF]'
2020-10-11 14:21:31   dev_node = '[UNDEF]'
2020-10-11 14:21:31   lladdr = '[UNDEF]'
2020-10-11 14:21:31   topology = 1
2020-10-11 14:21:31   ifconfig_local = '[UNDEF]'
2020-10-11 14:21:31   ifconfig_remote_netmask = '[UNDEF]'
2020-10-11 14:21:31   ifconfig_noexec = DISABLED
2020-10-11 14:21:31   ifconfig_nowarn = ENABLED
2020-10-11 14:21:31   ifconfig_ipv6_local = '[UNDEF]'
2020-10-11 14:21:31   ifconfig_ipv6_netbits = 0
2020-10-11 14:21:31   ifconfig_ipv6_remote = '[UNDEF]'
2020-10-11 14:21:31   shaper = 0
2020-10-11 14:21:31   mtu_test = 0
2020-10-11 14:21:31   mlock = DISABLED
2020-10-11 14:21:31   keepalive_ping = 0
2020-10-11 14:21:31   keepalive_timeout = 0
2020-10-11 14:21:31   inactivity_timeout = 0
2020-10-11 14:21:31   ping_send_timeout = 0
2020-10-11 14:21:31   ping_rec_timeout = 0
2020-10-11 14:21:31   ping_rec_timeout_action = 0
2020-10-11 14:21:31   ping_timer_remote = DISABLED
2020-10-11 14:21:31   remap_sigusr1 = 0
2020-10-11 14:21:31   persist_tun = ENABLED
2020-10-11 14:21:31   persist_local_ip = DISABLED
2020-10-11 14:21:31   persist_remote_ip = DISABLED
2020-10-11 14:21:31   persist_key = DISABLED
2020-10-11 14:21:31   passtos = DISABLED
2020-10-11 14:21:31   resolve_retry_seconds = 1000000000
2020-10-11 14:21:31   resolve_in_advance = ENABLED
2020-10-11 14:21:31   username = '[UNDEF]'
2020-10-11 14:21:31   groupname = '[UNDEF]'
2020-10-11 14:21:31   chroot_dir = '[UNDEF]'
2020-10-11 14:21:31   cd_dir = '[UNDEF]'
2020-10-11 14:21:31   writepid = '[UNDEF]'
2020-10-11 14:21:31   up_script = '[UNDEF]'
2020-10-11 14:21:31   down_script = '[UNDEF]'
2020-10-11 14:21:31   down_pre = DISABLED
2020-10-11 14:21:31   up_restart = DISABLED
2020-10-11 14:21:31   up_delay = DISABLED
2020-10-11 14:21:31   daemon = DISABLED
2020-10-11 14:21:31   inetd = 0
2020-10-11 14:21:31   log = DISABLED
2020-10-11 14:21:31   suppress_timestamps = DISABLED
2020-10-11 14:21:31   machine_readable_output = ENABLED
2020-10-11 14:21:31   nice = 0
2020-10-11 14:21:31   verbosity = 4
2020-10-11 14:21:31   mute = 0
2020-10-11 14:21:31   gremlin = 0
2020-10-11 14:21:31   status_file = '[UNDEF]'
2020-10-11 14:21:31   status_file_version = 1
2020-10-11 14:21:31   status_file_update_freq = 60
2020-10-11 14:21:31   occ = ENABLED
2020-10-11 14:21:31   rcvbuf = 0
2020-10-11 14:21:31   sndbuf = 0
2020-10-11 14:21:31   sockflags = 0
2020-10-11 14:21:31   fast_io = DISABLED
2020-10-11 14:21:31   comp.alg = 2
2020-10-11 14:21:31   comp.flags = 1
2020-10-11 14:21:31   route_script = '[UNDEF]'
2020-10-11 14:21:31   route_default_gateway = '[UNDEF]'
2020-10-11 14:21:31   route_default_metric = 0
2020-10-11 14:21:31   route_noexec = DISABLED
2020-10-11 14:21:31   route_delay = 0
2020-10-11 14:21:31   route_delay_window = 30
2020-10-11 14:21:31   route_delay_defined = DISABLED
2020-10-11 14:21:31   route_nopull = DISABLED
2020-10-11 14:21:31   route_gateway_via_dhcp = DISABLED
2020-10-11 14:21:31   allow_pull_fqdn = DISABLED
2020-10-11 14:21:31   route xx/255.255.255.0/vpn_gateway/default (not set)
2020-10-11 14:21:31   management_addr = '/data/user/0/de.blinkt.openvpn/cache/mgmtsocket'
2020-10-11 14:21:31   management_port = 'unix'
2020-10-11 14:21:31   management_user_pass = '[UNDEF]'
2020-10-11 14:21:31   management_log_history_cache = 250
2020-10-11 14:21:31   management_echo_buffer_size = 100
2020-10-11 14:21:31   management_write_peer_info_file = '[UNDEF]'
2020-10-11 14:21:31   management_client_user = '[UNDEF]'
2020-10-11 14:21:31   management_client_group = '[UNDEF]'
2020-10-11 14:21:31   management_flags = 16678
2020-10-11 14:21:31   shared_secret_file = '[UNDEF]'
2020-10-11 14:21:31   key_direction = 1
2020-10-11 14:21:31   ciphername = 'AES-256-CBC'
2020-10-11 14:21:31   ncp_enabled = ENABLED
2020-10-11 14:21:31   ncp_ciphers = 'AES-256-GCM:AES-128-GCM'
2020-10-11 14:21:31   authname = 'SHA512'
2020-10-11 14:21:31   prng_hash = 'SHA1'
2020-10-11 14:21:31   prng_nonce_secret_len = 16
2020-10-11 14:21:31   keysize = 0
2020-10-11 14:21:31   engine = DISABLED
2020-10-11 14:21:31   replay = ENABLED
2020-10-11 14:21:31   mute_replay_warnings = DISABLED
2020-10-11 14:21:31   replay_window = 64
2020-10-11 14:21:31   replay_time = 15
2020-10-11 14:21:31   packet_id_file = '[UNDEF]'
2020-10-11 14:21:31   test_crypto = DISABLED
2020-10-11 14:21:31   tls_server = DISABLED
2020-10-11 14:21:31   tls_client = ENABLED
2020-10-11 14:21:31   ca_file = '[INLINE]'
2020-10-11 14:21:31   ca_path = '[UNDEF]'
2020-10-11 14:21:31   dh_file = '[UNDEF]'
2020-10-11 14:21:31   cert_file = '[INLINE]'
2020-10-11 14:21:31   extra_certs_file = '[UNDEF]'
2020-10-11 14:21:31   priv_key_file = '[INLINE]'
2020-10-11 14:21:31   pkcs12_file = '[UNDEF]'
2020-10-11 14:21:31   cipher_list = 'TLS-DHE-RSA-WITH-AES-256-CBC-SHA256:TLS-DHE-RSA-WITH-AES-128-CBC-SHA256:TLS-DHE-DSS-WITH-AES-128-CBC-SHA256:TLS-DHE-RSA-WITH-AES-128-CBC-SHA'
2020-10-11 14:21:31   cipher_list_tls13 = '[UNDEF]'
2020-10-11 14:21:31   tls_cert_profile = '[UNDEF]'
2020-10-11 14:21:31   tls_verify = '[UNDEF]'
2020-10-11 14:21:31   tls_export_cert = '[UNDEF]'
2020-10-11 14:21:31   verify_x509_type = 1
2020-10-11 14:21:31   verify_x509_name = ''
2020-10-11 14:21:31   crl_file = '[UNDEF]'
2020-10-11 14:21:31   ns_cert_type = 0
2020-10-11 14:21:31   remote_cert_ku[i] = 65535
2020-10-11 14:21:31   remote_cert_ku[i] = 0
2020-10-11 14:21:31   remote_cert_ku[i] = 0
2020-10-11 14:21:31   remote_cert_ku[i] = 0
2020-10-11 14:21:31   remote_cert_ku[i] = 0
2020-10-11 14:21:31   remote_cert_ku[i] = 0
2020-10-11 14:21:31   remote_cert_ku[i] = 0
2020-10-11 14:21:31   remote_cert_ku[i] = 0
2020-10-11 14:21:31   remote_cert_ku[i] = 0
2020-10-11 14:21:31   remote_cert_ku[i] = 0
2020-10-11 14:21:31   remote_cert_ku[i] = 0
2020-10-11 14:21:31   remote_cert_ku[i] = 0
2020-10-11 14:21:31   remote_cert_ku[i] = 0
2020-10-11 14:21:31   remote_cert_ku[i] = 0
2020-10-11 14:21:31   remote_cert_ku[i] = 0
2020-10-11 14:21:31   remote_cert_ku[i] = 0
2020-10-11 14:21:31   remote_cert_eku = 'TLS Web Server Authentication'
2020-10-11 14:21:31   ssl_flags = 0
2020-10-11 14:21:31   tls_timeout = 2
2020-10-11 14:21:31   renegotiate_bytes = -1
2020-10-11 14:21:31   renegotiate_packets = 0
2020-10-11 14:21:31   renegotiate_seconds = 3600
2020-10-11 14:21:31   handshake_window = 60
2020-10-11 14:21:31   transition_window = 3600
2020-10-11 14:21:31   single_session = DISABLED
2020-10-11 14:21:31   push_peer_info = DISABLED
2020-10-11 14:21:31   tls_exit = DISABLED
2020-10-11 14:21:31   tls_crypt_v2_metadata = '[UNDEF]'
2020-10-11 14:21:31   server_network = 0.0.0.0
2020-10-11 14:21:31   server_netmask = 0.0.0.0
2020-10-11 14:21:31   server_network_ipv6 = ::
2020-10-11 14:21:31   server_netbits_ipv6 = 0
2020-10-11 14:21:31   server_bridge_ip = 0.0.0.0
2020-10-11 14:21:31   server_bridge_netmask = 0.0.0.0
2020-10-11 14:21:31   server_bridge_pool_start = 0.0.0.0
2020-10-11 14:21:31   server_bridge_pool_end = 0.0.0.0
2020-10-11 14:21:31   ifconfig_pool_defined = DISABLED
2020-10-11 14:21:31   ifconfig_pool_start = 0.0.0.0
2020-10-11 14:21:31   ifconfig_pool_end = 0.0.0.0
2020-10-11 14:21:31   ifconfig_pool_netmask = 0.0.0.0
2020-10-11 14:21:31   ifconfig_pool_persist_filename = '[UNDEF]'
2020-10-11 14:21:31   ifconfig_pool_persist_refresh_freq = 600
2020-10-11 14:21:31   ifconfig_ipv6_pool_defined = DISABLED
2020-10-11 14:21:31   ifconfig_ipv6_pool_base = ::
2020-10-11 14:21:31   ifconfig_ipv6_pool_netbits = 0
2020-10-11 14:21:31   n_bcast_buf = 256
2020-10-11 14:21:31   tcp_queue_limit = 64
2020-10-11 14:21:31   real_hash_size = 256
2020-10-11 14:21:31   virtual_hash_size = 256
2020-10-11 14:21:31   client_connect_script = '[UNDEF]'
2020-10-11 14:21:31   learn_address_script = '[UNDEF]'
2020-10-11 14:21:31   client_disconnect_script = '[UNDEF]'
2020-10-11 14:21:31   client_config_dir = '[UNDEF]'
2020-10-11 14:21:31   ccd_exclusive = DISABLED
2020-10-11 14:21:31   tmp_dir = '/data/data/de.blinkt.openvpn/cache'
2020-10-11 14:21:31   push_ifconfig_defined = DISABLED
2020-10-11 14:21:31   push_ifconfig_local = 0.0.0.0
2020-10-11 14:21:31   push_ifconfig_remote_netmask = 0.0.0.0
2020-10-11 14:21:31   push_ifconfig_ipv6_defined = DISABLED
2020-10-11 14:21:31   push_ifconfig_ipv6_local = ::/0
2020-10-11 14:21:31   push_ifconfig_ipv6_remote = ::
2020-10-11 14:21:31   enable_c2c = DISABLED
2020-10-11 14:21:31   duplicate_cn = DISABLED
2020-10-11 14:21:31   cf_max = 0
2020-10-11 14:21:31   cf_per = 0
2020-10-11 14:21:31   max_clients = 1024
2020-10-11 14:21:31   max_routes_per_client = 256
2020-10-11 14:21:31   auth_user_pass_verify_script = '[UNDEF]'
2020-10-11 14:21:31   auth_user_pass_verify_script_via_file = DISABLED
2020-10-11 14:21:31   auth_token_generate = DISABLED
2020-10-11 14:21:31   auth_token_lifetime = 0
2020-10-11 14:21:31   auth_token_secret_file = '[UNDEF]'
2020-10-11 14:21:31   port_share_host = '[UNDEF]'
2020-10-11 14:21:31   port_share_port = '[UNDEF]'
2020-10-11 14:21:31   vlan_tagging = DISABLED
2020-10-11 14:21:31   vlan_accept = all
2020-10-11 14:21:31 Warte 0s Sekunden zwischen zwei Verbindungsversuchen
2020-10-11 14:21:31   vlan_pvid = 1
2020-10-11 14:21:31   client = ENABLED
2020-10-11 14:21:31   pull = ENABLED
2020-10-11 14:21:31   auth_user_pass_file = '[UNDEF]'
2020-10-11 14:21:31 OpenVPN 2.5-icsopenvpn [git:icsopenvpn/v0.7.18-0-ge824dfb8] arm64-v8a [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD] built on Sep 15 2020
2020-10-11 14:21:31 library versions: OpenSSL 1.1.1g  21 Apr 2020, LZO 2.10
2020-10-11 14:21:31 MANAGEMENT: Connected to management server at /data/user/0/de.blinkt.openvpn/cache/mgmtsocket
2020-10-11 14:21:31 MANAGEMENT: CMD 'version 3'
2020-10-11 14:21:31 MANAGEMENT: CMD 'hold release'
2020-10-11 14:21:31 MANAGEMENT: CMD 'bytecount 2'
2020-10-11 14:21:31 MANAGEMENT: CMD 'proxy NONE'
2020-10-11 14:21:31 MANAGEMENT: CMD 'state on'
2020-10-11 14:21:32 Outgoing Control Channel Authentication: Using 512 bit message hash 'SHA512' for HMAC authentication
2020-10-11 14:21:32 Incoming Control Channel Authentication: Using 512 bit message hash 'SHA512' for HMAC authentication
2020-10-11 14:21:32 LZO compression initializing
2020-10-11 14:21:32 Control Channel MTU parms [ L:1622 D:1140 EF:110 EB:0 ET:0 EL:3 ]
2020-10-11 14:21:32 Data Channel MTU parms [ L:1622 D:1450 EF:122 EB:406 ET:0 EL:3 ]
2020-10-11 14:21:32 Local Options String (VER=V4): 'V4,dev-type tun,link-mtu 1602,tun-mtu 1500,proto UDPv4,comp-lzo,keydir 1,auth SHA512,keysize 256,tls-auth,key-method 2,tls-client'
2020-10-11 14:21:32 Expected Remote Options String (VER=V4): 'V4,dev-type tun,link-mtu 1602,tun-mtu 1500,proto UDPv4,comp-lzo,keydir 0,auth SHA512,keysize 256,tls-auth,key-method 2,tls-server'
2020-10-11 14:21:32 TCP/UDP: Preserving recently used remote address: [AF_INET]XX
2020-10-11 14:21:32 Socket Buffers: R=[212992->212992] S=[212992->212992]
2020-10-11 14:21:32 MANAGEMENT: CMD 'needok 'PROTECTFD' ok'
2020-10-11 14:21:32 UDP link local: (not bound)
2020-10-11 14:21:32 UDP link remote: [AF_INET]XX:1194
2020-10-11 14:21:32 MANAGEMENT: >STATE:1602418892,WAIT,,,,,,
2020-10-11 14:21:32 MANAGEMENT: >STATE:1602418892,AUTH,,,,,,
2020-10-11 14:21:32 TLS: Initial packet from [AF_INET]XX:1194, sid=4cc31d40 38545cab
2020-10-11 14:21:32 VERIFY OK: depth=1, 
2020-10-11 14:21:32 VERIFY KU OK
2020-10-11 14:21:32 Validating certificate extended key usage
2020-10-11 14:21:32 ++ Certificate has EKU (str) TLS Web Server Authentication, expects TLS Web Server Authentication
2020-10-11 14:21:32 VERIFY EKU OK
2020-10-11 14:21:32 VERIFY X509NAME OK: 
2020-10-11 14:21:32 VERIFY OK: depth=0, 
2020-10-11 14:21:33 Control Channel: TLSv1.2, cipher TLSv1.2 DHE-RSA-AES256-SHA256, peer certificate: 4096 bit RSA, signature: RSA-SHA256
2020-10-11 14:21:33 [server] Peer Connection Initiated with [AF_INET]XX:1194
2020-10-11 14:21:34 MANAGEMENT: >STATE:1602418894,GET_CONFIG,,,,,,
2020-10-11 14:21:34 SENT CONTROL [server]: 'PUSH_REQUEST' (status=1)
2020-10-11 14:21:34 PUSH: Received control message: 'PUSH_REPLY,route 10.8.0.1,topology net30,ping 10,ping-restart 120,ifconfig 10.8.0.14 10.8.0.13'
2020-10-11 14:21:34 OPTIONS IMPORT: timers and/or timeouts modified
2020-10-11 14:21:34 OPTIONS IMPORT: --ifconfig/up options modified
2020-10-11 14:21:34 OPTIONS IMPORT: route options modified
2020-10-11 14:21:34 OPTIONS ERROR: failed to negotiate cipher with server.  Add the server's cipher ('AES-256-CBC') to --data-ciphers (currently 'AES-256-GCM:AES-128-GCM') if you want to connect to this server.
2020-10-11 14:21:34 ERROR: Failed to apply push options
2020-10-11 14:21:34 Failed to open tun/tap interface
2020-10-11 14:21:34 TCP/UDP: Closing socket
2020-10-11 14:21:34 SIGUSR1[soft,process-push-msg-failed] received, process restarting
2020-10-11 14:21:34 MANAGEMENT: >STATE:1602418894,RECONNECTING,process-push-msg-failed,,,,,
2020-10-11 14:21:34 Warte 2s Sekunden zwischen zwei Verbindungsversuchen
2020-10-11 14:21:36 MANAGEMENT: CMD 'hold release'
2020-10-11 14:21:36 MANAGEMENT: CMD 'proxy NONE'
2020-10-11 14:21:36 MANAGEMENT: CMD 'bytecount 2'
2020-10-11 14:21:36 MANAGEMENT: CMD 'state on'
2020-10-11 14:21:36 MANAGEMENT: CMD 'signal SIGINT'
2020-10-11 14:21:36 Outgoing Control Channel Authentication: Using 512 bit message hash 'SHA512' for HMAC authentication
2020-10-11 14:21:36 Incoming Control Channel Authentication: Using 512 bit message hash 'SHA512' for HMAC authentication
2020-10-11 14:21:36 SIGINT[hard,init_instance] received, process exiting
2020-10-11 14:21:36 MANAGEMENT: >STATE:1602418896,EXITING,init_instance,,,,,
```

**Comment by schwabe** [comment]
@omid-1885 update app and reimport your  config
